SSO-Auth/Api/SSOController.cs

```diff
@@ -122,69 +122,70 @@ public SSOController(ILogger<SSOController> logger, ISessionManager sessionManag
                 // Role processing
                 // The regex matches any "." not preceded by a "\": a.b.c will be split into a, b, and c, but a.b\.c will be split into a, b.c (after processing the escaped dots)
                 // We have to first process the RoleClaim string
-                string[] segments = Regex.Split(config.RoleClaim?.Trim(), "(?<!\\\\)\\.");
-                // Now we make sure that any escaped "."s ("\.") are replaced with "."
-                for (int i = 0; i < segments.Length; i++)
-                {
-                    segments[i] = segments[i].Replace("\\.", ".");
-                }
+                string[] segments = string.IsNullOrEmpty(config.RoleClaim) ? Array.Empty<string>() : Regex.Split(config.RoleClaim?.Trim(), "(?<!\\\\)\\.");
```

Comment:
```suggestion
                string[] segments = string.IsNullOrEmpty(config.RoleClaim) ? Array.Empty<string>() : Regex.Split(config.RoleClaim.Trim(), "(?<!\\\\)\\.");
```
The `config.RoleClaim` is non-null here, thus no need to use `?.`.